### PR TITLE
[builder] Upgrade to CELLxGENE schema 5.1

### PR DESCRIPTION
**Pull request by Bento007**
# Reason
- To support [cellxgene schema 5.1 ](https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema/5.1.0)
- https://github.com/chanzuckerberg/single-cell-curation/issues/908

# changes
- update CXG_SCHEMA_VERSION to 5.1
- update the cellxgene-ontology-guide version to pin to latest major version.

Blocking: https://github.com/chanzuckerberg/single-cell-data-portal/pull/7078

### CODE REVIEW

File: tools/cellxgene_census_builder/src/cellxgene_census_builder/build_soma/globals.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 CENSUS_SCHEMA_VERSION = "2.0.1"
 
-CXG_SCHEMA_VERSION = "5.0.0"  # the CELLxGENE schema version supported
+CXG_SCHEMA_VERSION = "5.1.0"  # the CELLxGENE schema version supported
```
Comment (prathapsridharan) [inline tools/cellxgene_census_builder/src/cellxgene_census_builder/build_soma/globals.py:16]: @ebezzi @atolopko-czi - Should `CENSUS_SCHEMA_VERSION` also be bumped? If so, is it a `patch` or `minor` version bump?

**Review by prathapsridharan — CHANGES_REQUESTED** [review]

File: tools/cellxgene_census_builder/src/cellxgene_census_builder/build_soma/globals.py (hunk 1 above)
Comment (ebezzi) [reply tools/cellxgene_census_builder/src/cellxgene_census_builder/build_soma/globals.py:16]: I would only bump if there are noticeable and traceable changes (e.g. major ontology changes, new/removed assays). We should be able to add some doc changes if we bump it.

Comment (pablo-gar) [reply tools/cellxgene_census_builder/src/cellxgene_census_builder/build_soma/globals.py:16]: Yes we need to bump.the minor version number, and see my comment.
https://github.com/chanzuckerberg/cellxgene-census/issues/1194#issuecomment-2166838094

File: tools/cellxgene_census_builder/pyproject.toml [hunk 2]
```diff
@@ -36,7 +36,7 @@ dependencies= [
     #    https://github.com/TileDB-Inc/TileDB/blob/dev/format_spec/FORMAT_SPEC.md
     "tiledbsoma==1.9.3",
     "cellxgene-census==1.12.0",
-    "cellxgene-ontology-guide==0.6.1",
+    "cellxgene-ontology-guide==1.*",
```
Comment (ebezzi) [inline tools/cellxgene_census_builder/pyproject.toml:39]: We should have this pinned - upgrading a dependency that can affect the data components of the builder should be tested every time.

**Review by ebezzi — CHANGES_REQUESTED** [review]
Requesting one change related to dependency pins. 

Also, is there a changelog for the 5.0 -> 5.1 schema bump? This will determine how thoroughly this change should be tested.

File: tools/cellxgene_census_builder/pyproject.toml (hunk 2 above)
Comment (Bento007) [reply tools/cellxgene_census_builder/pyproject.toml:39]:
```suggestion
    "cellxgene-ontology-guide==1.0.0",
```

**Comment by Bento007** [comment]
@ebezzi release https://github.com/chanzuckerberg/single-cell-curation/releases/tag/v5.1.0

**Review by pablo-gar — CHANGES_REQUESTED** [review]
This should not be merged to main, either we build on top of it or we merge to e bigger one. 

The full process is [here](https://github.com/chanzuckerberg/cellxgene-census/issues/1194#issue-2349810112)

**Review by ebezzi — APPROVED** [review]

**Review by pablo-gar — APPROVED** [review]

**Review by prathapsridharan — APPROVED** [review]

**Pull request merged**